src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Assemblies/Ecma/EcmaAssembly.ManifestResources.cs

```diff
@@ -25,7 +25,7 @@ public sealed override ManifestResourceInfo GetManifestResourceInfo(string resou
             {
                 // Must get resource info from other assembly, and OR in the contained in another assembly information
                 ManifestResourceInfo underlyingManifestResourceInfo = internalManifestResourceInfo.ReferencedAssembly.GetManifestResourceInfo(resourceName)!;
-                internalManifestResourceInfo.FileName = underlyingManifestResourceInfo.FileName;
+                internalManifestResourceInfo.FileName = underlyingManifestResourceInfo.FileName ?? string.Empty;
```

Comment:
FYI, this is a functional change.  It looks like there is a path where FileName could be null, and the destination doesn't like that.